server/src/main/java/org/elasticsearch/common/geo/GeometryFormat.java

```diff
@@ -41,4 +46,10 @@
      */
     XContentBuilder toXContent(ParsedFormat geometry, XContentBuilder builder, ToXContent.Params params) throws IOException;
 
+    /**
+     * Serializes the geometry into a standard Java object.
+     *
+     * For example, the GeoJson format returns the geometry as a map, while WKT returns a string.
+     */
+    Object toObject(ParsedFormat geometry);
```

Comment:
I wonder if somebody can come up with a better name here :) maybe to `toXContentObject()` or `toXConentValue()` or something like this. I feel like toObject is misleadingly generic here.

Reply:
I previously had a method called `toXContentMap` that @talevy and @nik9000 found confusing, because we often refer to xContent and maps as distinct representations. Perhaps a name like `toXContentAsObject` could work?

Reply:
haha, oi. since there are javadocs explaining it now, I drop my naming argument. I'm good with whatever sounds good to you!

Reply:
I cannot say I love `toXContentAsObject` but I like it the best comparing to all other versions.